### PR TITLE
Fix dragging through firelocks

### PR DESCRIPTION
**Pull request by monster860**
:cl: monster860
tweak: When an edge firelock closes, if it closes between someone and something they're dragging it automatically pulls the thing they're dragging through.
/:cl:

### CODE REVIEW

**Comment by Hopekz** [comment]
Yes finally speedmerge THIS IS WHAT IS NEEDED the answer to our prayers all hail #7418

**Comment by Nickvr628** [comment]
speedmerge

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -311,6 +311,23 @@
...
+	if(density)
+		return TRUE
+	if(operating || welded)
+		return
+	var/turf/T1 = get_turf(src)
+	var/turf/T2 = get_step(T1, dir)
+	for(var/mob/living/M in T1)
+		if(M.stat == CONSCIOUS && M.pulling && M.pulling.loc == T2)
+			to_chat(M, "<span class='notice'>You pull [M.pulling] through [src] right as it closes</span>")
+			M.pulling.forceMove(T1)
```
Comment (alexkar598) [inline code/game/machinery/doors/firedoor.dm]: Whoa whoa whoa, ripping people from chairs as the firelock closes?
```suggestion
			M.pulling.Move_Pulled(T1)
```

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -311,6 +311,23 @@
...
+	for(var/mob/living/M in T2)
+		if(M.stat == CONSCIOUS && M.pulling && M.pulling.loc == T1)
+			to_chat(M, "<span class='notice'>You pull [M.pulling] through [src] right as it closes</span>")
+			M.pulling.forceMove(T2)
```
Comment (alexkar598) [inline code/game/machinery/doors/firedoor.dm]:
```suggestion
			M.pulling.Move_Pulled(T2)
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]

**Comment by Darkstick** [comment]
s p e e d m e r g e

**Review by alexkar598 — CHANGES_REQUESTED** [review]
Make sure the pullers move_force is stronger than the pulled move_resist

**Pull request merged**